bee-ternary/src/num_types.rs

```diff
@@ -0,0 +1,143 @@
+use std::convert::TryFrom;
+use crate::raw::{RawEncoding, RawEncodingBuf};
+
+pub use crate::{
+    trit::{Trit, Utrit, Btrit, ShiftTernary},
+    t1b1::{T1B1, T1B1Buf},
+    TritBuf,
+};
+
+use std::convert::{From, Into};
+
+
+const RADIX : u8 = 3;
+const NUM_TRITS_FOR_NUM_TYPE : usize = 27;
+const MAX_TRITS_IN_I64 : usize = (0.63 * 64 as f32) as usize + 1; // (log2/log3)*64
+
+
+fn min_trits(value: u64) -> usize {
+    let mut num = 1;
+    let mut vp : u64 = 1;
+
+    let mut value = value;
+    let mut num = 1;
+    while value > vp {
+        vp = vp * RADIX as u64 + 1;
+        num = num + 1;
+    }
+
+    num
+}
+
+
+
+
+//TODO - generalize over all encodings
```

Comment:
I don't think all encodings should support this. Maybe just T1B1 for now and T3B1 later would be sufficient.

Reply:
Yeah, you are probably right